application/src/main/java/org/togetherjava/tjbot/commands/filesharing/ShareLongAttachmentsMessageListener.java

```diff
@@ -0,0 +1,174 @@
+            return;
+        }
+
+        event.getMessage()
+            .getAttachments()
+            .stream()
+            .filter(attachment -> extensions.contains(attachment.getFileExtension()))
+            .forEach(attachment -> processAttachment(event, attachment));
+    }
+
+    private void processAttachment(MessageReceivedEvent event, Message.Attachment attachment) {
+        attachment.retrieveInputStream()
+            .thenApply(this::readAttachment)
+            .thenApply(content -> uploadToSharingService(event, attachment, content))
+            .thenAccept(response -> sendMessage(event, attachment, response));
+    }
+
+    private String readAttachment(@NotNull InputStream stream) {
+        try {
+            byte[] bytes = stream.readAllBytes();
+            return new String(bytes, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private HttpResponse<String> uploadToSharingService(MessageReceivedEvent event,
+            Message.Attachment attachment, String content) {
+        GistFile file = new GistFile();
+        file.setContent(content);
+
+        String parsedContent;
+        try {
+            parsedContent = JSON.writeValueAsString(file);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse content!", e);
+        }
+
+        String fileName = attachment.getFileName();
+        String fileExtension = attachment.getFileExtension();
+
+        if (fileExtension == null || fileExtension.equals("txt")) {
+            fileExtension = "java";
+        } else if (fileExtension.equals("fxml")) {
+            fileExtension = "xml";
+        }
+
+        fileName = fileName.substring(0, fileName.lastIndexOf(".")) + "." + fileExtension;
+
+        String body = """
+                {
+                  "description": "%s",
+                  "public": false,
+                  "files": {
+                      "%s": %s
+                  }
+                }""".formatted(event.getAuthor(), fileName, parsedContent);
+
+
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(API))
+            .header("Accept", "application/json")
+            .header("Authorization", "token " + API_KEY)
+            .POST(HttpRequest.BodyPublishers.ofString(body))
+            .build();
+
+        try {
+            return CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void sendMessage(MessageReceivedEvent event, Message.Attachment attachment,
+            HttpResponse<String> response) {
+        if (response.statusCode() < 200 || response.statusCode() > 299) {
+            logger.warn("Gist API unexpected response: " + response.body());
```

Comment:
use the loggers formatter syntax instead: `warn("Gist API unexpectes response: {}", response.body())`. that way its computed lazy.